packages/demo-protobuf/src/demo.spec.ts

```diff
@@ -0,0 +1,151 @@
+/* eslint-disable @typescript-eslint/camelcase */
+import { assert } from "@cosmjs/utils";
+import protobuf from "protobufjs";
+
+import reflectionRoot from "./demo";
+import demoJson from "./demo.json";
+import demoProto from "./demo.proto";
+import { cosmos_sdk as cosmosSdk, google } from "./generated/codecimpl";
+
+type MsgDemo = {
+  readonly example: string;
+};
+
+const { Coin } = cosmosSdk.v1;
+const { TxBody } = cosmosSdk.tx.v1;
+const { MsgSend } = cosmosSdk.x.bank.v1;
+const { Any } = google.protobuf;
+
+function getTypeName(typeUrl: string): string {
+  const parts = typeUrl.split(".");
+  return parts[parts.length - 1];
+}
+
+describe("protobuf demo", () => {
+  it("works with generated static code", () => {
+    const coin = Coin.create({
+      denom: "ucosm",
+      amount: "1234567890",
+    });
+    const msgSend = MsgSend.create({
+      fromAddress: Uint8Array.from(Array.from({ length: 20 }, () => 1)),
+      toAddress: Uint8Array.from(Array.from({ length: 20 }, () => 2)),
+      amount: [coin],
+    });
+    const msgSendBytes = MsgSend.encode(msgSend).finish();
+    const msgSendWrapped = Any.create({
+      type_url: "/cosmos.bank.MsgSend",
+      value: msgSendBytes,
+    });
+    const txBody = TxBody.create({
+      messages: [msgSendWrapped],
+      memo: "Some memo",
+      timeoutHeight: 9999,
+      extensionOptions: [],
+    });
+    const txBodyBytes = TxBody.encode(txBody).finish();
+
+    // Deserialization
+    const txBodyDecoded = TxBody.decode(txBodyBytes);
+    const msg = txBodyDecoded.messages[0];
+    assert(msg.value);
```

Comment:
This is a bit of annoyance in protobuf: [default values](https://developers.google.com/protocol-buffers/docs/proto3#default_values) are ususally not encoded, such that undefined and empty become the same thing. For the well known types we use often (e.g. Any, TxBody), we could create some encoding/decoding wrappers that normalize undefined to an empty byte array.